ai-server/app/core/s3.py

```
@@ -0,0 +1,43 @@
+            Bucket=settings.AWS_S3_BUCKET_NAME,
+            Key=s3_key,
+            ExtraArgs={"ContentType": content_type}
+        )
+
+        s3_url = f"https://{settings.AWS_S3_BUCKET_NAME}.s3.{settings.AWS_REGION}.amazonaws.com/{s3_key}"
+        file_size = os.path.getsize(local_path)
+
+        return {"s3_url": s3_url, "file_size": file_size}
+
+    except (BotoCoreError, ClientError) as e:
+        raise AIServerError(f"S3 upload failed: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)